### ISSUE TITLE
`vertical_whitespace_between_cases` considers comments

### ISSUE THREAD

**Issue by jshier** [realm/SwiftLint#6412]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Since 0.63, `vertical_whitespace_between_cases` considers inline comments in whether a newline is needed. For example, previously, this produced no warning with the setting enabled, as the cases were handled in a single line.

```swift
switch myEnum {
case .a: print("a")
// Now we handle b.
case .b: print("b")
}
```
However, now it requires a newline before the comment, despite the fact that the structure of the actual code doesn't trigger the warning.

```swift
switch myEnum {
case .a: print("a")

// Now we handle b.
case .b: print("b")
}
```

It seems to me that only the structure of the switch itself should be considered when determining the newline requirement, as was the case before. If this was an intentional change, perhaps an additional parameter can be added for better control?

### Environment

* SwiftLint version: 0.63.0
* Xcode version: 26.2
* Installation method used: CocoaPods
* Configuration file: directly enabled the rule, no configuration yet.

**Comment by SimplyDanny** [comment]
When I refactored the rule I intentionally went with this style assuming that only empty lines can really separate cases from each other. Comments are treated as belonging to either the previous or the next case. Previously, code like

```swift
switch myEnum {
case .a: print("a")
// Now we handle b.
case .b: print("b")

case .c: print("c")
}
```

was considered valid which felt a bit odd to me.

So I wouldn't like to change the current default. But an option, as you suggest, shouldn't be a big deal in case the old behavior is really preferred.

**Comment by jshier** [comment]
I agree your example is inconsistent, but now the inconsistency has simply shifted to the next line.

```swift
switch myEnum {
case .a: print("a")

// Now we handle b.
case .b: print("b")
case .c: print("c")
}
```
This particular case is a bit odd, in that single line `case`s are allowed to use no spacing at all, even when the rule is enabled. For this specific special case, it seems like no spacing should be required either way.

Another, similar, axis of customization would be handling mixed single-line and multiline cases. I haven't tested under the new rules, but previously, only the actual multiline `case`s would require the extra line, so you get into odd scenarios with mixed spacing even without the comment.

```swift
switch myEnum {
case .a: 
  print("a")

case .b: print("b")
case .c: 
  print("c")
}
```

Being able to control whether enums must including spacing for all cases when any single case requires would be good as well.

**Comment by SimplyDanny** [comment]
The rule ignores consecutive trivial one-line cases, but once you put comments in between cases they are no longer considered "trivial" and the normal rules kick in enforcing an empty line.

That explains all the examples here. So forgetting about the old behavior, would an option disabling the special treatment of trivial cases be sufficient to reach consistency? Then there would always need to be either an empty line or no space at all between all cases (depends on the existing `separation` option). Comments would not be treated as empty lines.
